src/main/java/Ui.java

```diff
@@ -0,0 +1,132 @@
+import java.util.Scanner;
+
+/**
+ * Represents the user interface that interacts with the user and acts as a view.
+ */
+public class Ui {
+    /**
+     * The input channel.
+     */
+    Scanner scanner = new Scanner(System.in);
+
+    /**
+     * Displays a message indicating an error in loading the task list save file.
+     */
+    public void showLoadingError() {
+        String message = "Cannot find save file.\n"
+                + "Creating a new one, even if it's a bother";
+
+        say(message);
+    }
+
+    /**
+     * Displays the list of tasks given.
+     * @param tasks the list of tasks
+     */
+    public void showTasks(TaskList tasks) {
+        String message = "Here are your procrastinated tasks:\n"
+                + tasks;
+
+        say(message);
+    }
+
+    /**
+     * Displays the deleted task.
+     * @param task the deleted task
+     */
+    public void showDeletedTask(Task task) {
+        String message = "Giving up already? Removed this:\n\t" + task;
+
+        say(message);
+    }
+
+    /**
+     * Displays the task marked as completed.
+     * @param task the completed task
+     */
+    public void showDonetask(Task task) {
+        String message = "Yay. You've finally done this task:\n\t" + task;
+
+        say(message);
+    }
+
+    /**
```

Comment:
Nicely done! just a small pointer, I think there should be a line between the descriptions and the @param? Like you did in the .java files above!